lib/engine/game/g_1840/game.rb

```diff
@@ -24,26 +24,114 @@ class Game < Game::Base
+        SELL_BUY_ORDER = :sell_buy
         CURRENCY_FORMAT_STR = '%d'
 
-        BANK_CASH = 12_000
+        BANK_CASH = 99_999
 
-        CERT_LIMIT = { 3 => 20, 4 => 16, 5 => 13, 6 => 11 }.freeze
+        CERT_LIMIT = { 3 => 16, 4 => 14, 5 => 13, 6 => 12 }.freeze
 
-        STARTING_CASH = { 3 => 800, 4 => 600, 5 => 480, 6 => 400 }.freeze
+        STARTING_CASH = { 3 => 300, 4 => 260, 5 => 230, 6 => 200 }.freeze
+
+        ADDITIONAL_CASH = 350
+
+        MARKET_TEXT = {
+          par: 'City Corporation Par',
+          par_2: 'Major Corporation Par',
+        }.freeze
+
+        STOCKMARKET_COLORS = Base::STOCKMARKET_COLORS.merge(
+          par: :red,
+          par_2: :green,
+        ).freeze
+
+        PAR_RANGE = {
+          city: [65, 75, 85, 95],
+          major: [70, 80, 90, 100],
+        }.freeze
+
+        INITIAL_CITY_PAR = {
+          'W' => 95,
+          'V' => 85,
+          'G' => 75,
+          'D' => 65,
+        }.freeze
+
+        INITIAL_CITY_TOKENS = {
+          'W' => [
+            { coordinate: 'I1' },
+            { coordinate: 'I9' },
+            { coordinate: 'I15' },
+            { coordinate: 'F24' },
+          ],
+          'V' => [
+            { city_index: 1, coordinate: 'A17' },
+            { coordinate: 'A13' },
+            { coordinate: 'B10' },
+            {  coordinate: 'C7' },
+            {  coordinate: 'F6' },
+            {  coordinate: 'G3' },
+          ],
+          'G' => [
+            { coordinate: 'A17' },
+            { coordinate: 'D12' },
+            { coordinate: 'I11' },
+          ],
+          'D' => [
+            { city_index: 2, coordinate: 'A17' },
+            { coordinate: 'D22' },
+            { coordinate: 'E23' },
+            { city_index: 1, coordinate: 'F24' },
+          ],
+        }.freeze
+
+        def setup
+          @first_stock_round = true
+          @tram_corporations = @corporations.select { |item| item.type == :minor }
+          @city_corporations = @corporations.select { |item| item.type == :city }
+          @major_corporations = @corporations.select { |item| item.type == :major }
+                                .sort_by { rand }.first(@players.size)
+
+          @city_corporations.each do |corporation|
+            par_value = INITIAL_CITY_PAR[corporation.id]
+            price = @stock_market.par_prices.find { |p| p.price == par_value }
+            @stock_market.set_par(corporation, price)
+            corporation.ipoed = true
+
+            initial_coordinates_info = INITIAL_CITY_TOKENS[corporation.id]
+
+            initial_coordinates_info.each do |info|
+              token = corporation.find_token_by_type
+              city_index = info[:city_index] || 0
+              hex_by_id(info[:coordinate]).tile.cities[city_index].place_token(corporation, token,
+                                                                               check_tokenable: false)
+            end
+          end
+
+          @corporations.clear
```

Comment:
How does this work when a tram corp appears in an Action, since actions use corporations_by_id ?

Reply:
I don't really know. I'll fix it then if needed I guess 🤷 
But thanks for the info